### PR TITLE
Add KIND to the pipeline to make things shine

### PR DESCRIPTION
**Pull request by ivanilvescmp**
## KIND - Kubernetes in Docker. 

_Real Kubernetes inside CircleCI build, not a mocked fake._

https://github.com/kubernetes-sigs/kind

Kubernetes "pocket cluster" is what we need to reliably test how our Terraform module really works.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+          kind create cluster --config .circleci/kind-config.yaml
     - run:
-        name: Create Terraform execution plan
-        working_directory: ./ci
-        command: make tf-plan
+        name: Run remote Docker container with Terraform
+        background: true
+        command: docker run --rm --name=tf --entrypoint=sleep ${TF_DOCKER_IMAGE} 3600
+    - run:
+        name: Sleep 5 seconds because life is hard and full of terrors
```
Comment (zsolt-cmp) [inline .circleci/config.yml]: 😃

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,22 +1,43 @@
...
+        name: Copy project repository into Docker Terraform container
+        command: docker cp ./ tf:/root/project
+    - run:
+        name: Add KIND kubeconfig to Docker Terraform container
+        command: |
+          docker exec tf mkdir -p /root/.kube
+          docker cp $(kind get kubeconfig-path) tf:/root/.kube/config
+    - run:
+        name: Deploy Terraform code to provision Kubernetes resources
+        command: docker exec tf bash -c "cd /root/project/ci && make tf-deploy"
```
Comment (dpiddockcmp) [inline .circleci/config.yml]: There is another path without race conditions with sleeping:
```shell
docker create -v /root/project -v /root/.kube --name tf_volumes alpine:3.10.1 /bin/true
docker cp ./ tf_volumes:/root/project
docker cp $(kind get kubeconfig-path) tf_volumes:/root/.kube/config
docker run --rm --name=tf --volumes-from tf_volumes -w /root/project/ci --entrypoint make ${TF_DOCKER_IMAGE} tf-deploy
```

Comment (ivanilvescmp) [reply .circleci/config.yml]: Awesome point! 👏

File: .circleci/config.yml (hunk 1 above)
Comment (ivanilvescmp) [reply .circleci/config.yml]: 🎐

**Pull request merged**